### PR TITLE
Final pre-release changes

### PR DESCRIPTION
**Pull request by michael-valdron**
**Please specify the area for this PR**

registry operator

**What does does this PR do / why we need it**:

Provides all remaining required changes for the first pre-release of the devfile registry operator.

**Preview Release on OperatorHub**

1. Copy all contents of the [bundle CSV](https://github.com/michael-valdron/devfile-registry-operator/blob/chore/pre-release/bundle/manifests/registry-operator.clusterserviceversion.yaml)
2. Go to https://operatorhub.io/preview and paste all contents in the box
3. Click "Preview Operator"

**Which issue(s) this PR fixes**:

Fixes #?

devfile/api#1211

**PR acceptance criteria**:

- [x] Change Log
    - Are all your changes documented under the [upcoming release](../CHANGELOG.md#unreleased) entry?
- [x] Test Coverage 
    - Are your changes sufficiently tested, and are any applicable test cases added or updated to cover your changes?
- [x] Gosec scans 
  - Fix all MEDIUM and higher findings and/or annotate a finding per gosec instructions: https://github.com/securego/gosec#annotating-code to address why a finding is not a security issue

Documentation
- [x] Does the [registry operator documentation](https://github.com/devfile/devfile-web/tree/main/libs/docs/src/docs/no-version) need to updated with your changes?

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -9,13 +9,13 @@ Fixes #?
...
-    - Are all your changes documented under the [upcoming release](../CHANGELOG.md#unreleased) entry?
+    - Are all your changes documented under the [upcoming release](https://github.com/devfile/registry-operator/tree/main/CHANGELOG.md#unreleased) entry?
 - [ ] Test Coverage 
     - Are your changes sufficiently tested, and are any applicable test cases added or updated to cover your changes?
 - [ ] Gosec scans 
   - Fix all MEDIUM and higher findings and/or annotate a finding per gosec instructions: https://github.com/securego/gosec#annotating-code to address why a finding is not a security issue
 
 Documentation
-- [ ] Does the [registry operator documentation](https://github.com/devfile/devfile-web/tree/main/libs/docs/src/docs/no-version) need to updated with your changes?
+- [ ] Does the registry operator documentation need to updated with your changes?
```
Comment (feloy) [inline .github/PULL_REQUEST_TEMPLATE.md:19]:
```suggestion
- [ ] Does the registry operator documentation need to be updated with your changes?
```

File: DEVFILE_REGISTRY.md [hunk 2]
```diff
@@ -1,10 +1,12 @@
 # Devfile Registry
 
-The Devfile Registry custom resource allows you to create and manage your own index server and registry viewer.
+DevfileRegistry is a custom resource allows you to create and manage your own index server and registry viewer. In order to be added, the Devfile
```
Comment (feloy) [inline DEVFILE_REGISTRY.md:3]:
```suggestion
DevfileRegistry is a custom resource allowing you to create and manage your own index server and registry viewer. In order to be added, the Devfile 
```

File: DEVFILE_REGISTRY.md [hunk 3]
```diff
@@ -1,10 +1,12 @@
 # Devfile Registry
 
-The Devfile Registry custom resource allows you to create and manage your own index server and registry viewer.
+DevfileRegistry is a custom resource allows you to create and manage your own index server and registry viewer. In order to be added, the Devfile 
+Registry must be reachable, supports the Devfile v2.0 spec and above, and is not using the default namespace.
```
Comment (feloy) [inline DEVFILE_REGISTRY.md:4]:
```suggestion
Registry must be reachable, must support the Devfile v2.0 spec and above, and cannot be deployed to the default namespace.
```

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (michael-valdron) [reply .github/PULL_REQUEST_TEMPLATE.md]: Revisions under https://github.com/devfile/registry-operator/pull/47/commits/0b7c7c22e0f349498a1741b8452671b023c4cfe4

File: DEVFILE_REGISTRY.md (hunk 2 above)
Comment (michael-valdron) [reply DEVFILE_REGISTRY.md]: Revisions under https://github.com/devfile/registry-operator/pull/47/commits/0b7c7c22e0f349498a1741b8452671b023c4cfe4

File: DEVFILE_REGISTRY.md (hunk 3 above)
Comment (michael-valdron) [reply DEVFILE_REGISTRY.md]: Revisions under https://github.com/devfile/registry-operator/pull/47/commits/0b7c7c22e0f349498a1741b8452671b023c4cfe4

**Review by feloy — APPROVED** [review]

**Pull request merged**